openedx/core/djangoapps/user_api/accounts/serializers.py

```diff
@@ -219,14 +220,19 @@ def to_representation(self, user):  # lint-amnesty, pylint: disable=arguments-di
                     "secondary_email_enabled": True,
                 }
             )
+        if auto_generated_username_enabled():
+            data.update(
+                {
+                    "hide_username": True,
```

Comment:
please change the key. 
```suggestion
                    "auto_generated_username_enabled": True,
```